homeassistant/core.py

```diff
@@ -79,7 +82,7 @@ def valid_state(state: str) -> bool:
     return len(state) < 256
 
 
-def callback(func: Callable[..., T]) -> Callable[..., T]:
+def callback(func: CALLABLE_T) -> CALLABLE_T:
```

Comment:
It would be nice if we can make a 'callback' type that a function becomes after it goes through this function.

Reply:
mypy has trouble understanding decorators that change function type

Reply:
Can we have callback extend callable or is that also not done?

Reply:
Template objects like `CALLABLE_T` can't be subclassed.
Moreover seems like type classes like `Callable` can only reside in stubs.